fog/sample-paykit/src/client.rs

```diff
@@ -362,6 +360,232 @@ impl Client {
+            return Err(Error::SciExpired);
+        }
+
+        let merkle_root_block = 0u64;
+
+        // Update sci's merkle proofs
+        sci.tx_in.proofs.clear();
+        for (idx, result) in self
+            .fog_merkle_proof
+            .get_outputs(sci.tx_out_global_indices.clone(), merkle_root_block)?
+            .results
+            .iter()
+            .enumerate()
+        {
+            if result.index != sci.tx_out_global_indices[idx] {
+                panic!("unhandled: Server returned indices in an unexpected order");
+            }
+            match result.status() {
+                Err(err) => panic!(
+                    "unhandled: Server failed to compute a merkle proof: {}",
+                    err
+                ),
+                Ok(None) => panic!("unhandled: Server did not find one of the outputs we need"),
+                Ok(Some((tx_out, proof))) => {
+                    if sci.tx_in.ring[idx] != tx_out {
+                        return Err(Error::SciGlobalIndexTxOutMismatch(
+                            sci.tx_out_global_indices[idx],
+                        ));
+                    }
+                    sci.tx_in.proofs.push(proof);
+                }
+            }
+        }
+
+        // Aggregate total required outlay
+        let mut outlay: HashMap<TokenId, u64> = Default::default();
+        outlay.insert(fee.token_id, fee.value);
+        for req_amount in sci.required_output_amounts.iter() {
+            *outlay
+                .entry(TokenId::from(req_amount.token_id))
+                .or_default() += req_amount.value;
+        }
+
+        // Compute the leftover from the signed input
+        let leftover = {
+            let token_id = TokenId::from(sci.pseudo_output_amount.token_id);
+            let sci_change = outlay.get(&token_id).cloned().unwrap_or(0);
+            if sci.pseudo_output_amount.value < sci_change {
+                return Err(Error::SciUnprofitable);
+            }
+            let value = sci.pseudo_output_amount.value - sci_change;
+            // We will use outlay to search our own wallet for required amounts, but
+            // we don't need to do that for this token id because it is supplied by the SCI
+            outlay.remove(&token_id);
+            Amount::new(value, token_id)
+        };
+
+        // Make fog resolver
+        let fog_uris = (&[&self.account_key.change_subaddress()])
+            .iter()
+            .filter_map(|addr| addr.fog_report_url())
+            .map(FogUri::from_str)
+            .collect::<core::result::Result<Vec<_>, _>>()?;
+        let fog_responses = self
+            .fog_report_conn
+            .fetch_fog_reports(fog_uris.into_iter())?;
+        let fog_resolver = FogResolver::new(fog_responses, &self.fog_verifier)?;
+
+        let block_version = BlockVersion::try_from(self.tx_data.get_latest_block_version())?;
+
+        let change_destination = ReservedDestination::from(&self.account_key);
```

Comment:
I guess I went a little further -- since there's only one fog uri here, ours, we don't need iter, filtermap, and Vec, we can just get the fog report url from the account key.